core/src/main/scala/caliban/execution/Field.scala

```diff
@@ -152,6 +154,7 @@ object Field {
           if (checkDirectives(resolvedDirectives)) {
             val t     = innerType.possibleTypes
               .flatMap(_.find(_.name.exists(typeCondition.map(_.name).contains)))
+              .orElse(typeCondition.flatMap(typeName => rootType.types.get(typeName.name)))
```

Comment:
With this, is there a need for searching through `possibleTypes`?

Reply:
No, `rootTypes.types` should return all types however nested they are so the interface should be found.